client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -330,6 +337,21 @@ private void fetchUpdatesIfNeeded() {
             compactIfNeeded();
         }
     }
+
+    void updateConfigIfNeeded() {
+        sync.fetchUpdates();
+        ReaderGroupState state = sync.getState();
+        if (!updateConfigTimer.hasRemaining() && state.isUpdatingConfig()) {
+            log.debug("Update the group config");
+            ReaderGroupConfig controllerConfig = getThrowingException(controller.getReaderGroupConfig(sync.getSynchronizerScopeName(),
+                    sync.getSynchronizerStreamName().replace(READER_GROUP_STREAM_PREFIX, "")));
+            if (state.getConfig().getGeneration() < controllerConfig.getGeneration()) {
+                Map<SegmentWithRange, Long> segments = getSegmentsForStreams(controller, controllerConfig);
+                sync.updateStateUnconditionally(new ReaderGroupState.ReaderGroupStateInit(controllerConfig, segments, getEndSegmentsForStreams(controllerConfig), false));
+            }
+            updateConfigTimer.reset(UPDATE_CONFIG_WINDOW);
```

Comment:
This only resets the timer if a config update was actually performed. However the goal of the timer was to minimise work on a per-event bais. This prevents that because the timer will almost never be not expired if the config updates rarely.

Reply:
As you mentioned it would only reset if an update was done. I've separated the timer check and the `updatingConfig` check. So irrespective of whether the update happens or not the timer will reset itself when there is no time remaining.